regparser/tree/struct.py

```diff
@@ -134,6 +143,12 @@ def walk(node, fn):
     return results
 
 
+def filter_walk(node, fn):
+    """Perform fn for every node in the tree and return a list of nodes
+    on which the function returns truthy."""
+    return walk(node, lambda n: n if fn(n.label) else None)
```

Comment:
@cmc333333 Ha. I wrote this before I wrote the code and never re-visited it. Sorry that this has been so sloppy this time around. I am committing to doing better next time!